src/backend/core/cms/views.py

```diff
@@ -0,0 +1,190 @@
+        pages = Page.objects.all()
+        serializer = PagePolymorphicDashboardSerializer(pages, many=True)
+        print(serializer.data)
+        return Response(serializer.data)
+
+    def post(self, request):
+        """
+        Adds a new page
+        """
+        serializer = PagePolymorphicDashboardSerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=HTTP_201_CREATED)
+        print(serializer.errors)
+        return Response(serializer.errors, status=HTTP_400_BAD_REQUEST)
+
+
+class PageFrontendDashboardDetailView(RetrieveUpdateDestroyAPIView):
+    """
+    View for getting (by ID), updating and deleting pages.
+    """
+
+    permission_classes = (AllowAny,)
+    authentication_classes = []
+
+    queryset = PageFrontend.objects.all()
+    serializer_class = PageFrontendDashboardSerializer
+
+
+class PageCMSDashboardDetailView(RetrieveUpdateDestroyAPIView):
+    """
+    View for getting (by ID), updating and deleting pages.
+    """
+
+    permission_classes = (AllowAny,)
+    authentication_classes = []
+
+    queryset = PageCMS.objects.all()
+    serializer_class = PageCMSDashboardSerializer
+
+
+class PageCategoryDashboardView(APIView):
+    """
+    View for adding new page category and listing all categories
+    """
+
+    permission_classes = (AllowAny,)
+    authentication_classes = []
+
+    def get(self, request):
+        """
+        Gets all page category.
+        """
+        pages = PageCategory.objects.all()
+        serializer = PageCategoryDashboardSerializer(pages, many=True)
+        print(serializer.data)
+        return Response(serializer.data)
+
+    def post(self, request):
+        """
+        Adds a new page category.
+        """
+        serializer = PageCategoryDashboardSerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=HTTP_201_CREATED)
+        print(serializer.errors)
+        return Response(serializer.errors, status=HTTP_400_BAD_REQUEST)
+
+
+class PageCategoryDashboardDetailView(
+    RetrieveUpdateDestroyAPIView,
+):
+    """
+    View for getting (by ID), updating and deleting page categories.
+    """
+
+    permission_classes = (AllowAny,)
+    authentication_classes = []
+
+    queryset = PageCategory.objects.all()
+    serializer_class = PageCategoryDashboardSerializer
+
+
+class PageCategoryPagesDashboardView(APIView):
+    """
+    View for listing all pages in a category
+    """
+
+    permission_classes = (AllowAny,)
+    authentication_classes = []
+
+    def get(self, request, pk):
+        """
+        Gets all pages in a category.
+        """
+        raise NotImplementedError
+
+
+class PageCategoryTypeDashboardView(APIView):
+    """
+    View for listing all page types
+    """
+
+    permission_classes = (AllowAny,)
+    authentication_classes = []
+
+    def get(self, request):
```

Comment:
same as above

Reply:
Sorry accidentally resolved. I don't think ListAPIView will be able to provide us `post` method?